nxc/protocols/winrm.py

```diff
@@ -170,34 +175,56 @@ def laps_search(self, username, password, ntlm_hash, domain):
 
     def print_host_info(self):
         if self.args.domain:
-            self.logger.extra["protocol"] = "HTTP"
+            self.logger.extra["protocol"] = "HTTPS" if self.ssl else "HTTP"
             self.logger.display(self.endpoint)
         else:
             self.logger.extra["protocol"] = "SMB"
+            self.logger.extra["port"] = "445"
             self.logger.display(f"{self.server_os} (name:{self.hostname}) (domain:{self.domain})")
-            self.logger.extra["protocol"] = "HTTP"
+            self.logger.extra["protocol"] = "HTTPS" if self.ssl else "HTTP"
+            self.logger.extra["port"] = self.port
             self.logger.display(self.endpoint)
 
         if self.args.laps:
             return self.laps_search(self.args.username, self.args.password, self.args.hash, self.domain)
         return True
 
     def create_conn_obj(self):
-        endpoints = [
-            f"https://{self.host}:{self.args.port if self.args.port else 5986}/wsman",
-            f"http://{self.host}:{self.args.port if self.args.port else 5985}/wsman",
-        ]
-
-        for url in endpoints:
+        if self.is_link_local_ipv6:
+            self.logger.info("winrm not support link-local ipv6, exitting...")
+            return False
+        endpoints = {
+            "HTTP": {
+                "protocol": "http", 
+                "port": self.port[0],
```

Comment:
Why [0]? I don't think we should assume http is the first

Reply:
Don’t worried about that, so I change the - - port argument to tell the user how to set http & https port

Reply:
If user given one port, that will use [0] both http&https

Reply:
Let me see, did you think it will be check-proto https,http?

Reply:
Oh, I have idea, we can get the index number of check_proto

when port length not equal 1, port variable should equal to check_proto index

Reply:
self.port[self.check_proto.index(“http”)] if len(self.port) == 2 else self.port[0]

Reply:
Fixed!

Now the port will follow the `check_proto` sequence